DEVELOPMENT.md

```diff
@@ -0,0 +1,107 @@
+ * The package is provided as a simple compressed archive - no package managers are used. This makes installation simple, but the downside is that **before accepting a contribution adding a new tool, we need to consider how much larger it makes the overall package**.
+
+# Development
+
+## Build Prerequisites
+At present, all of the builds are run on their native platform (no cross-compiling).
+
+**Windows**
+
+ 1. Install MSYS2 - install options:
+    * Use [chocolately](https://chocolatey.org/) and run `choco install msys2`
+    * Install using instructions on [the MSYS2 website](https://www.msys2.org/)
+ 2. Run the MinGW64 bash environment `mingw64.exe` (not the MSYS2 or MinGW32 environments).
+
+**macOS**
+
+1. Install [Xcode 11.4.1](https://developer.apple.com/services-account/download?path=/Developer_Tools/Xcode_11.4.1/Xcode_11.4.1.xip) to `/Applications/Xcode_11.4.1.app` (other versions might work but this is untested and unsupported)
+
+*Note that this requires macOS 10.15.2+ as a build-time dependency. The resulting builds should work on versions as old as macOS 10.10*
+
+2. Install [Homebrew](https://brew.sh/)
+
+**Linux**
+ 1. Create an Ubuntu 20.04 environment:
+    * The `ubuntu:20.04` docker container is a good option and this is the approach the CI build uses
+    * A virtual machine or even WSL on Windows will work too
+    * Running on bare metal is of course fine too, but other versions of Ubuntu are untested and unsupported.
+
+*Note that Ubuntu 20.04 is only a build-time dependency. The resulting builds are intended to run on any Linux distro*
+
+ 2. Make sure the `sudo` package is installed (it is not installed by default in the docker image)
+
+## Running a Build
+
+**WARNING**: The scripts will attempt to automatically install build dependencies by default! If you don't want this you can disable it first with `export INSTALL_DEPS=0`.
+
+The details of the required dependencies will not be documented here, so it is recommended to just let the scripts handle it - you can see what will be installed in `scripts/install_dependencies.sh`
+
+Build:
+
+```
+bash build.sh <arch>
+```
+
+Clean:
+
+```
+bash clean.sh <arch>
+```
+
+Current architectures:
+* linux_x86_64
+* windows_amd64
+* darwin
+
+Final packages will be deployed in the `./_packages/build_<arch>/` directory.
+
+By default the scripts will not build nextpnr-ecp5. See [this section](#building-nextpnr-ecp5) for details on how to build this too.
+
+## Disabling parts of the build
+
+The build scripts define many variables that may be used to disable parts of the build during development - see `build.sh` for details.
+
+If you place a `.env` file in the root of the repo then the bash scripts will source it. You can use this as a convenient way to locally override these variables without accidentally committing changes (the file is already in `.gitignore`).
```

Comment:
Thanks, this is a great trick to be sure to not realize you forget passing `COMPILE_NEXTPNR_ECP5=1` after an hour of building :sweat_smile: